### PR TITLE
[Infra] [Security] Update Scala and packages dependencies - Follow up

### PR DESCRIPTION
**Pull request by felipepessoto**
#### Which Delta project/connector is this regarding?
- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [X] Kernel
- [X] Other (connector, examples, benchmark)

## Description
#2828 updated SBT version to Spark Delta. This is a follow up to update other projects.
- Update SBT to 1.9.9. [CVE-2023-46122](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-46122)

## How was this patch tested?
CI

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

**Comment by felipepessoto** [comment]
@scottsand-db

**Comment by felipepessoto** [comment]
Rebased

+ @allisonport-db

**Comment by felipepessoto** [comment]
@scottsand-db @allisonport-db

File: kernel/project/build.properties
Comment (allisonport-db) [inline kernel/project/build.properties:1]: Can we just remove this file? We use the root sbt build for kernel anyways

Comment (felipepessoto) [reply kernel/project/build.properties:1]: Done

**Review by allisonport-db — APPROVED** [review]

**Pull request merged**